### PR TITLE
Added answer only loss for prompt learning

### PR DESCRIPTION
**Pull request by vadam5**
Signed-off-by: Virginia Adams <vadams@nvidia.com>

Adds ability to calculate loss on answer portion of the prompt only for p-tuning/prompt-tuning.

**Collection**: BigNLP

Updated 
- GPTPromptLearningDataset class
- MegatronGPTPromptLearningModel class
- megatron_gpt_prompt_learning_config

# Usage
Same as before, but task templates in the config now have ``answer_only_loss`` and ``answer_field`` added to them.
They are added to the config like:

```python
- taskname: 'rte'
  prompt_template: '<|VIRTUAL_PROMPT_0|>{text}{answer}'
  total_virtual_tokens: 100 
  virtual_token_splits: [100]
  truncate_field: null
  answer_only_loss: True
  answer_field: 'answer'
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation? **(In separate PR)**
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
- [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml [hunk 1]
```diff
@@ -59,10 +59,13 @@ model:
     total_virtual_tokens: 80 # Sum of tokens in virtual_token_splits must add to this number. Can differ between new and existing tasks, but must match across all new tasks being tuned at the same time.
     virtual_token_splits: [60, 20] # number of virtual tokens to be inserted at each VIRTUAL PROMPT location, must add to total_virtual_tokens
     truncate_field: 'passage' # The {field} in the prompt template whose text will be truncated if the input is too long, if null, inputs that are too long will just be skipped.
+    answer_only_loss: True 
+    answer_field: 'answer'
```
Comment (yidong72) [inline examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml:63]: shouldn't `answer_only_loss` always be `True`? I suggest remove it from the configure and make `loss_mask` only care about `answer` loss.

**Review by yidong72 — COMMENTED** [review]
Only one comment. Make `answer_only_loss` always True and remove the configuration to reduce confusing.

File: examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml (hunk 1 above)
Comment (vadam5) [reply examples/nlp/language_modeling/conf/megatron_gpt_prompt_learning_config.yaml:63]: Before the refactor, answer only loss = True didn't always give the best results when the prompt was short

**Review by yidong72 — APPROVED** [review]
LGTM!

**Review by okuchaiev — DISMISSED** [review]

**Comment by okuchaiev** [comment]
@vadam5 please have a look at why CI fails

**Review by okuchaiev — APPROVED** [review]
